waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -795,36 +799,50 @@ proc validateMessage*(rlnPeer: WakuRLNRelay, msg: WakuMessage,
     # accept messages whose epoch is within +-MAX_EPOCH_GAP from the current epoch
     debug "invalid message: epoch gap exceeds a threshold", gap = gap,
         payload = string.fromBytes(msg.payload)
+    waku_rln_invalid_messages.inc(labelValues=["invalid_epoch"])
     return MessageValidationResult.Invalid
 
   if not rlnPeer.validateRoot(msg.proof.merkleRoot):
     debug "invalid message: provided root does not belong to acceptable window of roots", provided=msg.proof.merkleRoot, validRoots=rlnPeer.validMerkleRoots
+    waku_rln_invalid_messages.inc(labelValues=["invalid_root"])
     return MessageValidationResult.Invalid
 
   # verify the proof
   let
     contentTopicBytes = msg.contentTopic.toBytes
     input = concat(msg.payload, contentTopicBytes)
-    proofVerificationRes = rlnPeer.rlnInstance.proofVerify(input, msg.proof)
+
+  waku_rln_proof_verification.inc()
+  waku_rln_proof_verification_time.time:
```

Comment:
My understanding from the nim-metrics description of `time` is that it measures time in seconds. As verification is very fast and in the order of milliseconds, it may mean that the measured time is always 0. Wondering whether you have tested this?

Reply:
This is an oversight on my end, will resolve!

Reply:
Added a `granularTime` template that resolves this. Ideally, we would move this template up to a common utils file so that the other protocols can use it. - [e18de66](https://github.com/status-im/nwaku/pull/1185/commits/e18de6673aa6501443390d3ccef7ff4ef267db0d)